quickstart/101-virtual-network-manager-create-mesh/main.tf

```diff
@@ -58,14 +58,56 @@ resource "azurerm_network_manager_network_group" "network_group" {
   network_manager_id = azurerm_network_manager.network_manager_instance.id
 }
 
-# Add the three virtual networks to the network group as static members
+# Add three virtual networks to a network group as dynamic members with Azure Policy
 
-resource "azurerm_network_manager_static_member" "static_members" {
-  count = 3
+resource "random_pet" "network_group_policy_name" {
+  prefix = "network-group-policy"
+}
+
+resource "azurerm_policy_definition" "network_group_policy" {
+  name         = "${random_pet.network_group_policy_name.id}"
+  policy_type  = "Custom"
+  mode         = "Microsoft.Network.Data"
+  display_name = "Policy Definition for Network Group"
+
+  metadata = <<METADATA
+    {
+      "category": "Azure Virtual Network Manager"
+    }
+  METADATA
+
+  policy_rule = <<POLICY_RULE
+    {
+      "if": {
+        "allOf": [
+          {
+              "field": "type",
+              "equals": "Microsoft.Network/virtualNetworks"
+          },
+          {
+            "allOf": [
+              {
+              "field": "Name",
+              "contains": "vnet"
```

Comment:
This prerequisite seems too loose and might interfere other tests in our test subscription. I would suggest a more precise prerequisite, like `match`, and make vnet's prefix random, like:

```hcl
resource "random_string" "prefix" {
  length = 4
  special = false
  upper = false
}

resource "random_pet" "virtual_network_name" {
  prefix = "vnet${random_string.prefix.result}"
}
```

Then the `allof` could be:

```json
{
  "field": "Name",
  "match": "${random_pet.virtual_network_name.id}.+"
}
```